diff -uNr termux-packages/packages/ffmpeg/build.sh termux-packages.mod/packages/ffmpeg/build.sh
--- termux-packages/packages/ffmpeg/build.sh	2023-02-13 01:03:12.989571100 +0800
+++ termux-packages.mod/packages/ffmpeg/build.sh	2023-02-13 01:13:46.348873400 +0800
@@ -36,7 +36,6 @@
 	local _EXTRA_CONFIGURE_FLAGS=""
 	if [ $TERMUX_ARCH = "arm" ]; then
 		_ARCH="armeabi-v7a"
-		_EXTRA_CONFIGURE_FLAGS="--enable-neon"
 	elif [ $TERMUX_ARCH = "i686" ]; then
 		_ARCH="x86"
 		# Specify --disable-asm to prevent text relocations on i686,
diff -uNr termux-packages/packages/libx265/build.sh termux-packages.mod/packages/libx265/build.sh
--- termux-packages/packages/libx265/build.sh	2023-02-13 01:03:14.924262400 +0800
+++ termux-packages.mod/packages/libx265/build.sh	2023-02-13 01:13:46.402710000 +0800
@@ -26,7 +26,7 @@
 
 	if [ "$TERMUX_ARCH" = arm ]; then
 		# Follow termux_step_setup_toolchain.sh
-		ARM_ARGS+=(-march=armv7-a -mfpu=neon -mfloat-abi=softfp -mthumb)
+		ARM_ARGS+=(-march=armv7-a -mfpu=vfpv3-d16 -mfloat-abi=softfp -mthumb)
 	fi
 
 	TERMUX_PKG_SRCDIR="$TERMUX_PKG_SRCDIR/source"
diff -uNr termux-packages/packages/libx265/source-CMakeLists.txt.patch termux-packages.mod/packages/libx265/source-CMakeLists.txt.patch
--- termux-packages/packages/libx265/source-CMakeLists.txt.patch	2023-02-13 01:03:14.924262400 +0800
+++ termux-packages.mod/packages/libx265/source-CMakeLists.txt.patch	2023-02-13 01:13:46.439709200 +0800
@@ -40,7 +40,6 @@
 -            endif()
 -        endif()
 +        set(ARM_ARGS @ARM_ARGS@ -fPIC)
-+        add_definitions(-DHAVE_NEON)
      endif()
      add_definitions(${ARM_ARGS})
      if(FPROFILE_GENERATE)
diff -uNr termux-packages/packages/ndk-sysroot/build.sh termux-packages.mod/packages/ndk-sysroot/build.sh
--- termux-packages/packages/ndk-sysroot/build.sh	2023-02-13 01:03:16.079915200 +0800
+++ termux-packages.mod/packages/ndk-sysroot/build.sh	2023-02-13 01:17:22.981376000 +0800
@@ -75,7 +75,7 @@
 
 	cp $LIBATOMIC/libatomic.a $TERMUX_PKG_MASSAGEDIR/$TERMUX_PREFIX/lib/
 
-	cp toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/lib/$TERMUX_HOST_PLATFORM/libcompiler_rt-extras.a \
+	cp toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/lib/$TERMUX_HOST_PLATFORM/$TERMUX_PKG_API_LEVEL/libcompiler_rt-extras.a \
 		$TERMUX_PKG_MASSAGEDIR/$TERMUX_PREFIX/lib/
 	# librt and libpthread are built into libc on android, so setup them as symlinks
 	# to libc for compatibility with programs that users try to build:
@@ -86,7 +86,7 @@
 
 	mkdir -p $TERMUX_PKG_MASSAGEDIR/$TERMUX_PREFIX/lib
 	# clang 13 requires libunwind on Android.
-	cp toolchains/llvm/prebuilt/linux-x86_64/lib64/clang/14.0.7/lib/linux/$NDK_ARCH/libunwind.a \
+	cp toolchains/llvm/prebuilt/linux-x86_64/lib64/clang/12.0.9/lib/linux/$NDK_ARCH/libunwind.a \
 		$TERMUX_PKG_MASSAGEDIR/$TERMUX_PREFIX/lib
 	for lib in librt.so libpthread.so libutil.so; do
 		echo 'INPUT(-lc)' > $TERMUX_PKG_MASSAGEDIR/$TERMUX_PREFIX/lib/$lib
diff -uNr termux-packages/scripts/build/toolchain/termux_setup_toolchain_23c.sh termux-packages.mod/scripts/build/toolchain/termux_setup_toolchain_23c.sh
--- termux-packages/scripts/build/toolchain/termux_setup_toolchain_23c.sh	2023-02-13 01:03:25.515102700 +0800
+++ termux-packages.mod/scripts/build/toolchain/termux_setup_toolchain_23c.sh	2023-02-13 01:13:46.595710200 +0800
@@ -48,7 +48,7 @@
 		# https://developer.android.com/ndk/guides/standalone_toolchain.html#abi_compatibility:
 		# "We recommend using the -mthumb compiler flag to force the generation of 16-bit Thumb-2 instructions".
 		# With r13 of the ndk ruby 2.4.0 segfaults when built on arm with clang without -mthumb.
-		CFLAGS+=" -march=armv7-a -mfpu=neon -mfloat-abi=softfp -mthumb"
+		CFLAGS+=" -march=armv7-a -mfpu=vfpv3-d16 -mfloat-abi=softfp -mthumb"
 		LDFLAGS+=" -march=armv7-a"
 		export GOARCH=arm
 		export GOARM=7
