diff --binary -uNr termux-packages/packages/ffmpeg/build.sh termux-packages.mod/packages/ffmpeg/build.sh
--- termux-packages/packages/ffmpeg/build.sh	2022-08-13 19:14:23.652641947 +0800
+++ termux-packages.mod/packages/ffmpeg/build.sh	2022-08-13 22:20:57.201845537 +0800
@@ -18,7 +18,6 @@
 	local _EXTRA_CONFIGURE_FLAGS=""
 	if [ $TERMUX_ARCH = "arm" ]; then
 		_ARCH="armeabi-v7a"
-		_EXTRA_CONFIGURE_FLAGS="--enable-neon"
 	elif [ $TERMUX_ARCH = "i686" ]; then
 		_ARCH="x86"
 		# Specify --disable-asm to prevent text relocations on i686,
diff --binary -uNr termux-packages/packages/libx265/build.sh termux-packages.mod/packages/libx265/build.sh
--- termux-packages/packages/libx265/build.sh	2022-08-13 19:14:23.674642582 +0800
+++ termux-packages.mod/packages/libx265/build.sh	2022-08-13 22:20:57.201845537 +0800
@@ -26,7 +26,7 @@
 
 	if [ "$TERMUX_ARCH" = arm ]; then
 		# Follow termux_step_setup_toolchain.sh
-		ARM_ARGS+=(-march=armv7-a -mfpu=neon -mfloat-abi=softfp -mthumb)
+		ARM_ARGS+=(-march=armv7-a -mfpu=vfpv3-d16 -mfloat-abi=softfp -mthumb)
 	fi
 
 	TERMUX_PKG_SRCDIR="$TERMUX_PKG_SRCDIR/source"
diff --binary -uNr termux-packages/packages/libx265/source-CMakeLists.txt.patch termux-packages.mod/packages/libx265/source-CMakeLists.txt.patch
--- termux-packages/packages/libx265/source-CMakeLists.txt.patch	2022-08-07 14:07:03.220893071 +0800
+++ termux-packages.mod/packages/libx265/source-CMakeLists.txt.patch	2022-08-13 22:20:57.201845537 +0800
@@ -40,7 +40,6 @@
 -            endif()
 -        endif()
 +        set(ARM_ARGS @ARM_ARGS@ -fPIC)
-+        add_definitions(-DHAVE_NEON)
      endif()
      add_definitions(${ARM_ARGS})
      if(FPROFILE_GENERATE)
diff --binary -uNr termux-packages/packages/ndk-sysroot/build.sh termux-packages.mod/packages/ndk-sysroot/build.sh
--- termux-packages/packages/ndk-sysroot/build.sh	2022-08-13 22:02:26.004576507 +0800
+++ termux-packages.mod/packages/ndk-sysroot/build.sh	2022-08-13 22:26:41.567845862 +0800
@@ -40,7 +40,7 @@
 
 	cp $LIBATOMIC/libatomic.a $TERMUX_PKG_MASSAGEDIR/$TERMUX_PREFIX/lib/
 
-	cp $TERMUX_STANDALONE_TOOLCHAIN/sysroot/usr/lib/$TERMUX_HOST_PLATFORM/libcompiler_rt-extras.a $TERMUX_PKG_MASSAGEDIR/$TERMUX_PREFIX/lib/
+	cp $TERMUX_STANDALONE_TOOLCHAIN/sysroot/usr/lib/$TERMUX_HOST_PLATFORM/$TERMUX_PKG_API_LEVEL/libcompiler_rt-extras.a $TERMUX_PKG_MASSAGEDIR/$TERMUX_PREFIX/lib/
 	# librt and libpthread are built into libc on android, so setup them as symlinks
 	# to libc for compatibility with programs that users try to build:
 	local _SYSTEM_LIBDIR=/system/lib64
@@ -50,7 +50,7 @@
 	NDK_ARCH=$TERMUX_ARCH
 	test $NDK_ARCH == 'i686' && NDK_ARCH='i386'
 	# clang 13 requires libunwind on Android.
-	cp $TERMUX_STANDALONE_TOOLCHAIN/lib64/clang/14.0.6/lib/linux/$NDK_ARCH/libunwind.a .
+	cp $TERMUX_STANDALONE_TOOLCHAIN/lib64/clang/12.0.9/lib/linux/$NDK_ARCH/libunwind.a .
 
 	for lib in librt.so libpthread.so libutil.so; do
 		echo 'INPUT(-lc)' > $TERMUX_PKG_MASSAGEDIR/$TERMUX_PREFIX/lib/$lib
diff --binary -uNr termux-packages/packages/pypy/termux-build.patch termux-packages.mod/packages/pypy/termux-build.patch
--- termux-packages/packages/pypy/termux-build.patch	2022-08-07 14:07:03.221893086 +0800
+++ termux-packages.mod/packages/pypy/termux-build.patch	2022-08-13 22:20:57.208845740 +0800
@@ -197,7 +197,7 @@
 +
 +def _update_cflags_for_termux_arm(cflags):
 +    return ('-I' + TARGET_ROOTFS_BASE + '@TERMUX_PREFIX@/include',
-+            "-march=armv7-a", "-mfpu=neon",
++            "-march=armv7-a", "-mfpu=vfpv3-d16",
 +            "-mfloat-abi=softfp", "-mthumb", ) + cflags
 +
 +def _update_link_flags_for_termux_arm(ldflags):
diff --binary -uNr termux-packages/packages/pypy3/termux-build.patch termux-packages.mod/packages/pypy3/termux-build.patch
--- termux-packages/packages/pypy3/termux-build.patch	2022-08-07 14:07:03.221893086 +0800
+++ termux-packages.mod/packages/pypy3/termux-build.patch	2022-08-13 22:20:57.210845798 +0800
@@ -197,7 +197,7 @@
 +
 +def _update_cflags_for_termux_arm(cflags):
 +    return ('-I' + TARGET_ROOTFS_BASE + '@TERMUX_PREFIX@/include',
-+            "-march=armv7-a", "-mfpu=neon",
++            "-march=armv7-a", "-mfpu=vfpv3-d16",
 +            "-mfloat-abi=softfp", "-mthumb", ) + cflags
 +
 +def _update_link_flags_for_termux_arm(ldflags):
diff --binary -uNr termux-packages/scripts/build/toolchain/termux_step_setup_toolchain_23c.sh termux-packages.mod/scripts/build/toolchain/termux_step_setup_toolchain_23c.sh
--- termux-packages/scripts/build/toolchain/termux_step_setup_toolchain_23c.sh	2022-08-13 19:14:23.714643736 +0800
+++ termux-packages.mod/scripts/build/toolchain/termux_step_setup_toolchain_23c.sh	2022-08-13 22:20:57.211845827 +0800
@@ -48,7 +48,7 @@
 		# https://developer.android.com/ndk/guides/standalone_toolchain.html#abi_compatibility:
 		# "We recommend using the -mthumb compiler flag to force the generation of 16-bit Thumb-2 instructions".
 		# With r13 of the ndk ruby 2.4.0 segfaults when built on arm with clang without -mthumb.
-		CFLAGS+=" -march=armv7-a -mfpu=neon -mfloat-abi=softfp -mthumb"
+		CFLAGS+=" -march=armv7-a -mfpu=vfpv3-d16 -mfloat-abi=softfp -mthumb"
 		LDFLAGS+=" -march=armv7-a"
 		export GOARCH=arm
 		export GOARM=7
diff --binary -uNr termux-packages/scripts/properties.sh termux-packages.mod/scripts/properties.sh
--- termux-packages/scripts/properties.sh	2022-08-13 19:14:23.714643736 +0800
+++ termux-packages.mod/scripts/properties.sh	2022-08-13 22:20:57.212845856 +0800
@@ -5,8 +5,8 @@
 
 TERMUX_SDK_REVISION=8512546
 TERMUX_ANDROID_BUILD_TOOLS_VERSION=30.0.3
-TERMUX_NDK_VERSION_NUM=25
-TERMUX_NDK_REVISION=""
+TERMUX_NDK_VERSION_NUM=23
+TERMUX_NDK_REVISION="c"
 TERMUX_NDK_VERSION=$TERMUX_NDK_VERSION_NUM$TERMUX_NDK_REVISION
 # when changing the above:
 # remove TERMUX_PKG_REVISION in:
